src/ClientResumeStatusCallback.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+
+namespace reactivesocket {
+
+class ClientResumeStatusCallback {
+public:
+
+  // Called when a RESUME_OK frame is received during resuming operation
+  virtual void onResumeOk() = 0;
+
+  // Called when an ERROR frame with CONNECTION_ERROR is received during resuming operation
+  virtual void onConnectionError(folly::exception_wrapper ex) = 0;
```

Comment:
should the destructor automatically call onConnectionError if onResumeOk or onConnectionError were never called?